drivers/Watchdog.h

```diff
@@ -0,0 +1,174 @@
+ *  malfunctions.
+ *
+ * Example:
+ * @code
+ *
+ * Watchdog watchdog = Watchdog();
+ * watchdog.set_timeout(2000);
+ * watchdog.start();
+ *
+ * while (true) {
+ *    watchdog.kick();
+ *
+ *    // Application code
+ * }
+ * @endcode
+ *
+ */
+class Watchdog
+{
+private:
+  uint32_t timeout_;
+  uint32_t window_;
+  bool enable_sleep_;
+
+public:
+    /** Create a watchdog timer with no configured parameters
+     */
+    Watchdog() : timeout_(0), window_(0), enable_sleep_(true) {}
+
+
+    /** Create a Watchdog timer with specified parameters
+     *
+     *  @param timeout Timeout of the watchdog in milliseconds
+     *  @param window  Time period of the window of the watchdog [default: 0]
+
+     *  @note
+     *    Setting window to anything other than 0 will enable windowed mode for
+     *    the watchdog.
+     *
+     *    The window value must be less than the timeout value.
+     */
+    Watchdog(const uint32_t timeout, const uint32_t window = 0)
+        : timeout_(timeout), window_(window), enable_sleep_(true) {}
+
+
+public:
+    /** Start the watchdog with the given configuration
+     *
+     * @return status  WATCHDOG_STATUS_OK if the watchdog timer was started
+     *                 successfully. WATCHDOG_INVALID_ARGUMENT if one of the input
+     *                 parameters is out of range for the current platform.
+     *                 WATCHDOG_NOT_SUPPORTED if one of the enabled input
+     *                 parameters is not supported by the current platform.
+     */
+    watchdog_status_t start();
+
+
+    /** Refreshes the watchdog timer.
+     *
+     * This function should be called periodically before the watchdog times out.
+     * Otherwise, the system is reset.
+     *
+     * If the watchdog timer is not currently running this function does nothing
+     */
+    void kick();
+
+
+    /** Stops the watchdog timer
+     *
+     * Calling this function will attempt to disable any currently running
+     * watchdog timers if supported by the current platform.
+     *
+     * @return Returns WATCHDOG_STATUS_OK if the watchdog timer was succesfully
+     *         stopped, or if the timer was never started. Returns
+     *         WATCHDOG_STATUS_NOT_SUPPORTED if the watchdog cannot be disabled
+     *         on the current platform.
+     */
+    watchdog_status_t stop();
+
+
+    /** Get the watchdog timer refresh value
+     *
+     * This function returns the refresh timeout of the watchdog timer.
+     *
+     * @return Reload value for the watchdog timer in milliseconds.
+     */
+    uint32_t reload_value() const;
+
+
+    /** Get the maximum refresh value for the current platform in milliseconds
+     *
+     * @return Maximum refresh value supported by the watchdog for the current
+     *         platform in milliseconds
+     */
+    uint32_t max_timeout() const;
+
+
+    /** Set the timeout/refresh value of the watchdog in milliseconds
```

Comment:
Do we need that? I think that makes it more complicated with questions about what happens after the `start` is called.